### PR TITLE
Test against LCG nightlies

### PR DESCRIPTION
**Pull request by andresailer**
This adds builds against the dev3 and dev4 nightly builds of LCG, which are allowed to fail, but could at least give an early warning if something will break in the new ROOT version

Also adds a job to test `BUILD_SHARED_LIBS=OFF` in view of #579

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,12 +7,19 @@ services:
...
 env:
-  matrix:
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96python3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
-    - COMPILER=clang; LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=clang8; PYTHON_CHECK="no"
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=14; COMPILER_VERSION=gcc62; PYTHON_CHECK="no"
-    - PYTHON_CHECK="yes"; COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8;
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"; CMAKE_ARGS="-DBUILD_SHARED_LIBS=OFF"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96python3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
```
Comment (petricm) [inline .travis.yml]:
```suggestion
  - COMPILER=gcc;   LCG_RELEASE=LCG_96bpython3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
```

File: .travis.yml [hunk 2]
```diff
@@ -7,12 +7,19 @@ services:
 language: cpp
 
 env:
-  matrix:
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96python3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
-    - COMPILER=clang; LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=clang8; PYTHON_CHECK="no"
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=14; COMPILER_VERSION=gcc62; PYTHON_CHECK="no"
-    - PYTHON_CHECK="yes"; COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8;
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
```
Comment (petricm) [inline .travis.yml]:
```suggestion
  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc9; PYTHON_CHECK="no"
```

File: .travis.yml [hunk 3]
```diff
@@ -7,12 +7,19 @@ services:
...
+  - COMPILER=clang; LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=clang8; PYTHON_CHECK="no"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=14; COMPILER_VERSION=gcc62; PYTHON_CHECK="no"
+  - COMPILER=gcc; LCG_RELEASE=dev3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+  - COMPILER=gcc; LCG_RELEASE=dev4; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+  - PYTHON_CHECK="yes"; COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8;
+
+matrix:
+  allow_failures:
+    - env: COMPILER=gcc; LCG_RELEASE=dev3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+    - env: COMPILER=gcc; LCG_RELEASE=dev4; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
```
Comment (petricm) [inline .travis.yml]: One of these can also be gcc9

File: .travis.yml (hunk 1 above)
Comment (petricm) [reply .travis.yml]: I have not previously changed this, because the python3 stack was not published till recently

File: .travis.yml [hunk 4]
```diff
@@ -7,12 +7,19 @@ services:
...
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96python3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
-    - COMPILER=clang; LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=clang8; PYTHON_CHECK="no"
-    - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=14; COMPILER_VERSION=gcc62; PYTHON_CHECK="no"
-    - PYTHON_CHECK="yes"; COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8;
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"; CMAKE_ARGS="-DBUILD_SHARED_LIBS=OFF"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96bpython3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
+  - COMPILER=clang; LCG_RELEASE=LCG_96b; STANDARD=17; COMPILER_VERSION=clang8; PYTHON_CHECK="no"
+  - COMPILER=gcc;   LCG_RELEASE=LCG_96b; STANDARD=14; COMPILER_VERSION=gcc62; PYTHON_CHECK="no"
+  - COMPILER=gcc; LCG_RELEASE=dev3; STANDARD=17; COMPILER_VERSION=gcc8; PYTHON_CHECK="no"
```
Comment (andresailer) [inline .travis.yml]:
```suggestion
  - COMPILER=gcc; LCG_RELEASE=dev3; STANDARD=17; COMPILER_VERSION=gcc9; PYTHON_CHECK="no"
```

**Comment by andresailer** [comment]
There is no LCIO for the gcc9 option

**Pull request merged**